tests/jmap.rs

```diff
@@ -0,0 +1,50 @@
+#![cfg(feature = "invocation")]
+
+extern crate error_chain;
+extern crate jni;
+
+use jni::objects::{
+    JMap,
+    JObject,
+};
+
+mod util;
+use util::{
+    attach_current_thread,
+    unwrap,
+};
+
+#[test]
+pub fn jmap_push_and_iterate() {
+    let env = attach_current_thread();
+    let data = &["hello", "world", "from", "test"];
+
+    let map_object = unwrap(&env, env.new_object("java/util/HashMap", "()V", &[]));
```

Comment:
If the test requires a particular order of keys to simplify comparison, I'd suggest either:
1. Using [`java/util/TreeMap`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/TreeMap.html) — it keeps keys ordered.
2. Or, better, [`LinkedHashMap`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/LinkedHashMap.html) — it preserves the insertion order, hence neither the source nor the `collected` need to be sorted below.

Reply:
The test does not require any particular order. I just tried to replicate the error I had seen on Android and this was with a `HashMap`.